### PR TITLE
feat: added Python API

### PR DESCRIPTION
**Pull request by camilovelezr**
This PR introduces a new Python API for WIC.

Purpose of the API: to provide programmatic access for building workflows.

Usage:
Assume the user wants to create a two step workflow. The user already has one CLT for each step.
```python3
from pathlib import Path
from wic import Step, Workflow
step1 = Step("/home/user/clt1.cwl")
step2 = Step("/home/user/clt2.cwl")

step1.inpDir = Path("/home/user/input")
step1.darkField = True
step2.inpDir = step1.outDir # explicit link

workflow = Workflow([step1, step2], "two_step_workflow", path=Path("/home/user/two_step_workflow"))
workflow.compile()
workflow.run() # optional -> will use cwltool
```
`Workflow.compile()` will:
1. Generate the relevant WIC YAML - defining which tools, in what order, with what parameters - and dump it to disk.
2. Call WIC from the command line to compile the YAML that was just dumped to disk

`Workflow.run()` will run the workflow generated by WIC, specifying the inputs YML generated by WIC, and using `cwltool` with `--relax-path-checks`

### CODE REVIEW

**Review by Nicholas-Schaub — APPROVED** [review]

File: src/wic/api/_types.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+CWL_TYPES_DICT: dict[str, object] = {
+    "null": None,
+    "boolean": bool,
+    "int": int,
+    "long": int,
+    "float": float,
+    "double": float,
+    "string": str,
+    # "File": CWLFile,
+    # "Directory": CWLDir,
```
Comment (hamshkhawar) [inline src/wic/api/_types.py:27]: Could you please remove duplicated keys and values from CWL_TYPES_DICT dictionary?

File: src/wic/api/pythonapi.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        print(f"Compiling {self.name}")
+        proc = subprocess.run(
+            args=["wic", "--yaml", self.yml_path],
+            capture_output=True,
+            cwd=self.path,
+            check=True,
+            text=True,
+            universal_newlines=True,
+        )
+        print(proc.stdout)
```
Comment (hamshkhawar) [inline src/wic/api/pythonapi.py:347]: Replace "print" with by using logger.info() as you already using logging package

**Review by hamshkhawar — COMMENTED** [review]
@camilovelezr I have added couple of comments on this PR. I haven't tested this code? How can I do it?

File: src/wic/api/_types.py (hunk 1 above)
Comment (camilovelezr) [reply src/wic/api/_types.py:27]: Addressed in #135

File: src/wic/api/pythonapi.py (hunk 2 above)
Comment (camilovelezr) [reply src/wic/api/pythonapi.py:347]: Addressed in #135

**Comment by camilovelezr** [comment]
> @camilovelezr I have added couple of comments on this PR. I haven't tested this code? How can I do it?

@hamshkhawar Thank you Hamdah, I addressed your comments and responded. I will soon add unit tests

**Pull request merged**